src/App.tsx

```diff
@@ -1,16 +1,83 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import debounce from 'debounce';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface Props {
+  debounceDelay?: number;
+}
+
+export const App: React.FC<Props> = ({ debounceDelay = 300 }) => {
+  const [query, setQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<{
+    name: string;
+    born: number;
+    died: number;
+  } | null>(null);
+
+  const filteredPeople = useMemo(() => {
+    if (debouncedQuery.trim() === '') {
+      return peopleFromServer;
+    }
+
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(debouncedQuery.toLowerCase()),
+    );
+  }, [debouncedQuery]);
+
+  const onFocus = () => {
+    if (query.trim() === '') {
+      setDebouncedQuery('');
+    }
+  };
+
+  const debouncedSetQuery = useMemo(
+    () => debounce(setDebouncedQuery, debounceDelay),
+    [debounceDelay],
+  );
+
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const value = event.target.value;
+
+      setQuery(value);
+      debouncedSetQuery(value);
+
+      if (
+        value === '' ||
+        !filteredPeople.find(person => person.name === value)
+      ) {
+        setSelectedPerson(null);
+      }
+    },
+    [debouncedSetQuery, filteredPeople],
+  );
+
+  const handlePersonSelect = (person: {
+    name: string;
+    born: number;
+    died: number;
+  }) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+  };
+
+  const displayedPerson =
+    selectedPerson || (filteredPeople.length > 0 ? filteredPeople[0] : null);
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+        {selectedPerson ? (
+          <h1 className="title" data-cy="title">
+            {`${displayedPerson?.name} (${displayedPerson?.born} - ${displayedPerson?.died})`}
+          </h1>
+        ) : (
+          <h1 className="title" data-cy="title">
+            No selected person
+          </h1>
```

Comment:
h1 is the same in each case, let's just change content inside h1

Reply:
done